src/Cli/dotnet/Commands/Run/RunCommandSelector.cs

```diff
@@ -0,0 +1,436 @@
+            return selectedFramework != null;
+        }
+        else
+        {
+            Reporter.Error.WriteLine(string.Format(CliCommandStrings.RunCommandExceptionUnableToRunSpecifyFramework, "--framework"));
+            Reporter.Error.WriteLine();
+            Reporter.Error.WriteLine(CliCommandStrings.RunCommandAvailableTargetFrameworks);
+            Reporter.Error.WriteLine();
+
+            for (int i = 0; i < frameworks.Length; i++)
+            {
+                Reporter.Error.WriteLine($"  {i + 1}. {frameworks[i]}");
+            }
+
+            Reporter.Error.WriteLine();
+            Reporter.Error.WriteLine($"{CliCommandStrings.RunCommandExampleText}: dotnet run --framework {frameworks[0]}");
+            Reporter.Error.WriteLine();
+            selectedFramework = null;
+            return false;
+        }
+    }
+
+    /// <summary>
+    /// Prompts the user to select a target framework from the available options using Spectre.Console.
+    /// </summary>
+    private static string? PromptForTargetFramework(string[] frameworks)
+    {
+        try
+        {
+            var prompt = new SelectionPrompt<string>()
+                .Title($"{CyanMarkup}{Markup.Escape(CliCommandStrings.RunCommandSelectTargetFrameworkPrompt)}{EndMarkup}")
+                .PageSize(10)
+                .MoreChoicesText($"{GrayMarkup}({Markup.Escape(CliCommandStrings.RunCommandMoreFrameworksText)}){EndMarkup}")
+                .AddChoices(frameworks)
+                .EnableSearch()
+                .SearchPlaceholderText(CliCommandStrings.RunCommandSearchPlaceholderText);
+
+            return Spectre.Console.AnsiConsole.Prompt(prompt);
+        }
+        catch (Exception)
+        {
+            // If Spectre.Console fails (e.g., terminal doesn't support it), return null
+            return null;
+        }
+    }
+
+    /// <summary>
+    /// Represents a device item returned from the ComputeAvailableDevices MSBuild target.
+    /// </summary>
+    public record DeviceItem(string Id, string? Description, string? Type, string? Status, string? RuntimeIdentifier);
+
+    /// <summary>
+    /// Computes available devices by calling the ComputeAvailableDevices MSBuild target if it exists.
+    /// </summary>
+    /// <param name="devices">List of available devices if the target exists, null otherwise</param>
+    /// <returns>True if the target was found and executed, false otherwise</returns>
+    public bool TryComputeAvailableDevices(out List<DeviceItem>? devices)
+    {
+        devices = null;
+
+        if (!OpenProjectIfNeeded(out var projectInstance))
+        {
+            // Invalid project file, return false
+            return false;
+        }
+
+        // Check if the ComputeAvailableDevices target exists
+        if (!projectInstance.Targets.ContainsKey(Constants.ComputeAvailableDevices))
+        {
+            return false;
+        }
+
+        // Build the target
+        var buildResult = projectInstance.Build(
```

Comment:
Not a blocker for this work, commenting so I can easily raise a linked issue:

With these kinds of 'invoke a target to get info' scenarios (`ComputeAvailableDevices` and `ComputeRunArguments`) we potentially wait for an undetermined amount of time. We need to have a more user-friendly experience here:
* if the operation takes more than X time, give the user some signal that we're doing work on their behalf
* potentially enforce some kind of maximal allowed time limit to prevent infinite waits